### ISSUE TITLE
Add Support for Humidifier LUH-M101S-WUS

### ISSUE THREAD

**Issue by sjafferali** [webdjoe/pyvesync#177]
Would be great if this supported the humidifier LUH-M101S-WUS. 

I tried performing a packet capture using charles proxy per https://github.com/webdjoe/pyvesync#feature-requests on both an iphone and mac, but am experiencing issues after enabling SSL proxying for https://smartapi.vesync.com . Specifically, no requests on the application succeed and charles proxy only captures a bunch of CONNECT failed requests with the error "Input Error: EOF". 

Is there something additional that needs to be done for the app to work other than enabling SSL proxying for smartapi.vesync.com ? I have already imported and trusted the Charles Proxy root certificates on my trust stores. 

Would be happy to provide captures if you can provide any additional guidance on configuring charles proxy to work with the vesync app.

**Comment by webdjoe** [comment]
I'm guessing VeSync started ssl pinning on iOS which is a real bummer. If you've followed the directions, installed the trusted certificate and enabled SSL proxying, the only other options are figuring out a way to bypass SSL pinning on an android emulator with frida and objection (Tried this in the past and was not successful but there may be some new objection recipes out) or a combination of decompiling the APK to look for the endpoints and testing calls with Insomnia or Postman based on other similar devices. I would try a few other apps with all domains enabled for SSL proxying first and make sure it is working as intended. Does the app connect or it says there's been an error?

**Comment by sjafferali** [comment]
> I'm guessing VeSync started ssl pinning on iOS which is a real bummer. 

This is what I was afraid of as well. 

> I would try a few other apps with all domains enabled for SSL proxying first and make sure it is working as intended. 

I tried a few different proxying apps, with the same results unfortunately. 

> Does the app connect or it says there's been an error?

It fails to connect with "Network not available". If I start proxying after the devices are already loaded, I get "Oops something went wrong. Please check your internet connection or contact support."

**Comment by webdjoe** [comment]
Don't know if this works but may be worth a shot: #180

**Comment by sjafferali** [comment]
@webdjoe . Unfortunately, I do not have an android device to try this on. I wonder if there is something similar for iOS.

**Comment by webdjoe** [comment]
Not that I'm aware of but you can use an Android emulator, such as Android Studio using a generic version of the Android OS.

**Comment by M1K3SD1** [comment]
@sjafferali I can try and help with the captures if you'd like. I was successful with my Air Purifiers as you can see here: https://github.com/webdjoe/pyvesync/issues/181. It's not easy at all and I'll need your cooperation if you want, but it can be done.

Since I have a packet capturing environment ready but no access to such device, I can suggest you creating a temporary account in vesync with a temporary email, connect your device to this account.

Send me these credentials privately so I can connect and capture the packets, after which I will log out and you can disconnect you device from this temp account.

Note that this will give me access to your actual device, so if you're not comfortable with this don't do this.

**Comment by sjafferali** [comment]
@M1K3SD1 Sure. I tried using an emulator, but ran into quite a few issues myself.

I would be happy to create an account for you, if you think you can help. How do you prefer I send you the credentials privately?

**Comment by M1K3SD1** [comment]
Yeah, I spent quite a few days on setting it up and figuring out how to unpin the SSL.
You can send it to my mail (mikemisc84 at gmail)

**Comment by sjafferali** [comment]
@M1K3SD1 Sent. Let me know if you need anything else.

**Comment by M1K3SD1** [comment]
@sjafferali just your approval to operate this device remotely to capture its traffic please.

**Comment by sjafferali** [comment]
@M1K3SD1 Yes, feel free and thank you for helping with this!

**Comment by M1K3SD1** [comment]
Captured all functions, cleaned the sensitive information, and sent it to @webdjoe.
Thank you, for now, you can disconnect the device from this account.

**Comment by sjafferali** [comment]
Thank you. @M1K3SD1 !

**Comment by webdjoe** [comment]
In the process of merging in releasing, thank you @sjafferali  for the PR with tests and @M1K3SD1 for the assist!

**Comment by dag81** [comment]
Could any one possibly post the response payload for what this returns on calling getHumidifierStatus - the response object? Sorry to be a pain just want to cross check if pos as the API's have all updated? Just looking for the "result" object that has the full json structure returned including sample data? @webdjoe any chance you can help if pos?

**Comment by sjafferali** [comment]
@dag81 this is the full response body.
```
{
    "traceId": "1686282712315",
    "code": 0,
    "msg": "request success",
    "module": null,
    "stacktrace": null,
    "result": {
        "traceId": "1686282712315",
        "code": 0,
        "result": {
            "powerSwitch": 1,
            "humidity": 59,
            "targetHumidity": 50,
            "virtualLevel": 1,
            "mistLevel": 1,
            "workMode": "manual",
            "waterLacksState": 0,
            "waterTankLifted": 0,
            "autoStopSwitch": 1,
            "autoStopState": 0,
            "screenSwitch": 0,
            "screenState": 0,
            "scheduleCount": 0,
            "timerRemain": 0,
            "errorCode": 0
        }
    }
}
```

I also noticed it was drastically different than anything else that this library currently supports. I still have the captures so lmk if you need anything else.

**Comment by dag81** [comment]
@sjafferali, Thank you so much for providing :) I can't see any errors in PyVeSyncs implementation 👍  It does confirm what I suspect however which is that the protocol's have changed to be now camelCase with more efficient numeric based variables. This looks very much to be the same as the Air Purifiers Vital models, so I am guessing going forwards this is the new format that is being used.

**Comment by dag81** [comment]
Hi @sjafferali / anyone else, sorry just in case you know. I am trying to figure out these bits to the new protocol:

In result we have:

timerRemain - Is this in mins of the current timer active?
scheduleCount - Current schedules the device is using?

More importantly:

I am presuming ....State = the current state, whereas .....Switch = the configured state.

E.g. screenSwitch = Whether the display is generally configured as off
screenState = Whether the display is currently active (maybe because of user interactions?)

Could anyone confirm if my assumptions above are correct, or completely wrong and correct them? Any idea's much appreciated.

**Comment by Crepetsky** [comment]
Just got one of these, and would love to see support added as well. I'm able to assist with access to the unit for testing, if needed.

Keep up the good work!

**Comment by webdjoe** [comment]
This has been released in 2.1.10. Please file another issue you come across any bugs. Thank you @sjafferali for the PR!
